File: go/solutions/01-beginner/ex024_matrix_transpose/matrix_transpose.go
// Package matrixtranspose — Exercise 024 (reference solution).
package matrixtranspose

// Transpose returns the transpose of the rectangular matrix m.
func Transpose(m [][]int) [][]int {
	if len(m) == 0 {
		return [][]int{}
	}
	rows := len(m)
	cols := len(m[0])

	result := make([][]int, cols)
	for c := 0; c < cols; c++ {
		result[c] = make([]int, rows)
		for r := 0; r < rows; r++ {
			result[c][r] = m[r][c]
		}
	}
	return result
}

File: php/exercises/01-beginner/ex029_matrix_transpose/MatrixTranspose.php
<?php

declare(strict_types=1);

namespace FeWoLearning\Exercises\Beginner\Ex029MatrixTranspose;

/*
Exercise 029 - Matrix transpose (beginner).

Goal:   Transpose a rectangular matrix represented as a list of row arrays.
Drills: nested arrays, the array_map(null, ...$matrix) transpose idiom.
Passes: MatrixTransposeTest
*/
final class MatrixTranspose
{
    private function __construct()
    {
    }

    public static function transpose(array $matrix): array
    {
        throw new \RuntimeException('TODO');
    }
}
